repotracker/repotracker.go

```diff
@@ -105,6 +105,10 @@ func (repoTracker *RepoTracker) FetchRevisions(ctx context.Context, numNewRepoRe
 	}
 
 	if lastRevision == "" {
+		numRevisions := numNewRepoRevisionsToFetch
```

Comment:
You already have access to `settings.RepoTracker.NumNewRepoRevisionsToFetch` from the first line of `FetchRevisions`, so I think you can eliminate the `numNewRepoRevisionsToFetch` argument to `FetchRevisions` altogether.